src/lib/Imp.hs

```diff
@@ -480,6 +495,11 @@ makeDest nameHint destType = do
               let dcs' = applyDataDefParams def params
               contents <- forM dcs' $ \(DataConDef _ bs) -> forM (toList bs) (rec . binderType)
               return $ Con $ SumAsProd ty tag contents
+        RecordTy types -> Record <$> forM types rec
+        VariantTy types -> do
+          tag <- rec IntTy
+          contents <- forM (toList types) rec
+          return $ Con $ SumAsProd ty tag $ map (\x->[x]) contents
```

Comment:
This looks incorrect to me, because it doesn't preserve the type!

Reply:
It saves the type into the `SumAsProd` constructor, which I think preserves the type for any downstream usage? At least the typechecker doesn't complain about it.

Reply:
`SumAsProd` was intended for ADTs (and if it passes the type checker then that was probably an oversight on my part). But this seems like a fine use of it too. Btw, we're going to remove it soon in favor of a `Minicase Var [AtomAlt]` atom, as mentioned in #179.